ReactiveCocoa/DynamicProperty.swift

```diff
@@ -37,7 +37,13 @@ public final class DynamicProperty<Value>: MutablePropertyProtocol {
 	///              Most UI controls are not!
 	public var producer: SignalProducer<Value?, NoError> {
 		return (object.map { $0.reactive.values(forKeyPath: keyPath) } ?? .empty)
-			.map { $0 as! Value }
```

Comment:
The subtle thing here is there's a difference between `PropertyProtocol.Value` (associated type) and `DynamicProperty<Value>` (generic type parameter). It's really unclear, but `Value` in the context of this dynamic cast is actually `Value?`, whereas the same name in the context of the type declaration is `DynamicProperty.Value` (non-optional).

If it was possible to namespace the type names you'd have something like `PropertyProtocol.Value == DynamicProperty.Value?`.

I believe the only way to cause this to actually crash would be to pass an object of an unexpected type into `setValue(_:forKey:)`, or if an object of unexpected type was assigned from Objective-C.

The type parameter naming could certainly be improved.